mypipes_t.py

```diff
@@ -0,0 +1,242 @@
+        
+        return x[self.feature_names]
+
+    def get_feature_names(self) :
+                
+        return self.feature_names
+
+class DataFrameImputer(BaseEstimator,TransformerMixin):
+    
+    def __init__(self):
+        
+        
+        self.feature_names=[]
+        self.impute_dict={}
+        
+    
+    def fit(self,x,y=None):
+        
+        self.feature_names=x.columns
+        
+        for col in x.columns:
+            
+            if x[col].dtype=='O':
+                self.impute_dict[col]='missing'
+            else :
+                self.impute_dict[col]=x[col].median()
+                
+        return self
+    
+    def transform(self,x) :
+        
+        return x.fillna(self.impute_dict)
+        
+    def get_feature_names(self) :
+                
+        return self.feature_names
+    
+class convert_to_datetime(BaseEstimator,TransformerMixin):
+    
+    def __init__(self):
+        
+        self.feature_names=[]
+        
+    
+    def fit(self,x,y=None):
+
+        self.feature_names=x.columns
+
+        return self 
+    
+    def transform(self,x) :
+
+        for col in x.columns:
+
+            x[col]=pd.to_datetime(x[col],errors='coerce')
+
+        return x
+    
+    def get_feature_names(self) :
+                
+        return self.feature_names
+    
+class creat_dummies(BaseEstimator,TransformerMixin):
+    
+    def __init__(self,freq_cutoff=0):
+        
+        
+        self.freq_cutoff=freq_cutoff
+        self.var_cat_dict={}
+        self.feature_names=[]
+        
+    
+    def fit(self,x,y=None):
+
+        for col in x.columns:
+
+            k=x[col].value_counts()
+            cats=k.index[k>=self.freq_cutoff][:-1]
+
+            self.var_cat_dict[col]=list(cats)
+
+        for col in self.var_cat_dict.keys():
+
+            for cat in self.var_cat_dict[col]:
+                self.feature_names.append(col+'_'+str(cat))
+
+        return self 
+    
+    def transform(self,x) :
+        
+        dummy_data=x.copy()
+
+        for col in self.var_cat_dict.keys():
+
+            for cat in self.var_cat_dict[col]:
+
+                name=col+'_'+str(cat)
+
+                dummy_data[name]=(dummy_data[col]==cat).astype(int)
+
+            del dummy_data[col]
+
+        return dummy_data
+
+    def get_feature_names(self) :
+                
+        return self.feature_names
+
+class vectorize(BaseEstimator,TransformerMixin):
+    
+     def __init__(self):
+        
+        self.feature_names=['Consumer complaint narrative']
+        
+        
+     def fit(self,x,y=None):
+                   
+        return self
+    
+     def transform(self,x):
+         
+         vector=TfidfVectorizer(analyzer='word',stop_words='english',max_df=0.8,min_df=0.01,max_features=200)
+        
+         
+         x['Consumer complaint narrative']=dx['Consumer complaint narrative'].str.replace('XXXX','')
+         X = vector.fit_transform(x['Consumer complaint narrative'])

```

Comment:
fit_transform is not right, this will. make it fit on every new data that you try to transform